### PR TITLE
[feature](Nereids): use session variable to enable rule

### PR DESCRIPTION
**Pull request by jackwener**
## Proposed changes

use session variable to enable rule

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 1]
```diff
@@ -2285,6 +2288,14 @@ public Set<Integer> getDisableNereidsRules() {
                 .collect(ImmutableSet.toImmutableSet());
     }
 
+    public Set<Integer> getEnableNereidsRules() {
+        return Arrays.stream(enableNereidsRules.split(",[\\s]*"))
+                .filter(rule -> !rule.isEmpty())
+                .map(rule -> rule.toUpperCase(Locale.ROOT))
+                .map(rule -> RuleType.valueOf(rule).type())
+                .collect(ImmutableSet.toImmutableSet());
+    }
```
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:2297]: Add MemCache

**Review by keanji-x — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**